### PR TITLE
Fanout language updates

### PR DESCRIPTION
**Pull request by mgloystein**

### CODE REVIEW

**Comment by brekk** [comment]
Seems probably pretty good. but I'm no go expert.

File: utils.go [hunk 1]
```diff
@@ -257,3 +258,39 @@ func FisherYatesShuffle(inputs []interface{}, nhash ...uint64) {
...
+}
+
+func delveCreate(obj map[string]interface{}, key string, val interface{}) bool {
+	props := strings.Split(key, ".")
+
+	if len(props) == 0 {
+		return false
+	} else if len(props) == 1 {
+		obj[key] = val
+		return true
```
Comment (Luo-Jia) [inline utils.go:283]: line 279-283 can be condensed into:

```
if len(props) < 2 {
  return (len(props) == 1)
}
```

File: utils.go [hunk 2]
```diff
@@ -257,3 +258,39 @@ func FisherYatesShuffle(inputs []interface{}, nhash ...uint64) {
...
+	} else {
+		var result interface{}
+
+		result, ok := obj[props[0]]
+		if !ok {
```
Comment (Luo-Jia) [inline utils.go]: fun syntax:

```
if result, ok := obj[props[0]]; !ok {
...
```

File: utils.go (hunk 1 above)
Comment (mgloystein) [reply utils.go:283]: need the assignment too...

**Comment by sheeley** [comment]
👍

**Pull request merged**